### PR TITLE
allow missing CVR headers

### LINKED ISSUE #813: CSV Candidate Auto-Load fails if no text in upper left corner
The following CSV will fail auto-load candidates:

```
,A,B,C,D
Person 1,1,2,3,4
```

The following will succeed:
```
Name,A,B,C,D
Person 1,1,2,3,4
```

### PR DESCRIPTION
**Pull request by artoonie**
closes #813 

This type of CSV is allowed and I don't see a reason to not support it. Added a test to account for it.

Because `getHeaderNames` skips all null/empty columns, I had to move away from using the built-in header parser.

I also put the `Generic CSV Writer` behind a `encounteredSourceProblem` flag since it will lead to more exceptions on failure.

### CODE REVIEW

**Comment by yezr** [comment]
Do we need to then add some basic validation checks to make sure that each candidate column has characters? I created a .csv [generic_csv_test_cvr_2.csv](https://github.com/BrightSpots/rcv/files/14960483/generic_csv_test_cvr_2.csv) that has a blank candidate name and I get the following error. 

image

**Comment by artoonie** [comment]
Good catch -- yes, will need to fix that.

**Comment by artoonie** [comment]
Fixed -- it took a bit more work than expected, the built-in header parser really doesn't like null header names, even though it's fairly common practice in CVRs/Spreadsheets to have empty headers...go ahead and give this a whirl?

**Comment by yezr** [comment]
image
success!

File: src/test/resources/network/brightspots/rcv/test_data/generic_csv_test/generic_csv_test_cvr.csv [hunk 1]
```diff
@@ -1,4 +1,4 @@
-Response ID,Lettuce,Broccoli,Cucumber,Cauliflower,Write-In
+,Lettuce,Broccoli,Cucumber,Cauliflower,Write-In
```
Comment (HEdingfield) [inline src/test/resources/network/brightspots/rcv/test_data/generic_csv_test/generic_csv_test_cvr.csv:1]: Can we please keep this as-is, and make a new file and test for this case instead?

File: src/main/java/network/brightspots/rcv/TabulatorSession.java [hunk 2]
```diff
@@ -344,16 +344,18 @@ private List<CastVoteRecord> parseCastVoteRecords(ContestConfig config) {
...
+                new ResultsWriter().setContestConfig(config).setTimestampString(timestampString);
+        this.convertedFilePath =
+                writer.writeRctabCvrCsv(
+                        castVoteRecords,
+                        perSourceDataForCsv,
+                        config.getOutputDirectory());
+      } catch (IOException exception) {
+        // error already logged in ResultsWriter
+      }
     }
```
Comment (HEdingfield) [inline src/main/java/network/brightspots/rcv/TabulatorSession.java:359]: Nit: `else` this with line 361.

File: src/main/java/network/brightspots/rcv/CsvCvrReader.java
Comment (HEdingfield) [inline src/main/java/network/brightspots/rcv/CsvCvrReader.java:1]: @tarheel would you mind taking a look at this too?

File: src/main/java/network/brightspots/rcv/TabulatorSession.java [hunk 3]
```diff
@@ -392,8 +392,18 @@ private LoadedCvrData parseCastVoteRecords(ContestConfig config, Progress progre
       progress.markFileRead();
     }
 
-    // Output the RCTab-CSV CVR
-    if (!encounteredSourceProblem) {
+    if (encounteredSourceProblem) {
```
Comment (HEdingfield) [inline src/main/java/network/brightspots/rcv/TabulatorSession.java:399]: Looks like you've got duplicate code between here and line 420.

Comment (artoonie) [reply src/main/java/network/brightspots/rcv/TabulatorSession.java:399]: Bad merge! Fixed, thanks for catching.

File: src/main/java/network/brightspots/rcv/CsvCvrReader.java
Comment (HEdingfield) [reply src/main/java/network/brightspots/rcv/CsvCvrReader.java:1]: I feel confident enough to approve this without @tarheel's input now. 👍

**Review by HEdingfield — APPROVED** [review]

**Pull request merged**